lib/session.js

```diff
@@ -177,13 +191,14 @@ exports.persist = function (req, res, next) {
 		process.exit(1);
 	}
 	if (keystone.get('cookie signin') && !req.session.userId && req.signedCookies['keystone.uid'] && req.signedCookies['keystone.uid'].indexOf(':') > 0) {
-		exports.signin(req.signedCookies['keystone.uid'], req, res, function () {
-			next();
-		}, function (err) {
-			res.clearCookie('keystone.uid');
-			req.user = null;
-			next();
+		var cookieOpts = _.defaults({}, keystone.get('cookie signin options'), {
+			signed: true,
+			httpOnly: true,
 		});
+		cookieOpts.maxAge = 0;
+		res.clearCookie('keystone.uid', cookieOpts);
```

Comment:
According to [documentation](https://expressjs.com/en/4x/api.html#res.clearCookie), it says:

> Web browsers and other compliant clients will only clear the cookie if the given options is identical to those given to res.cookie(), excluding expires and maxAge.

Do we need to keep `maxAge` as same as the `maxAge` given to `res.cookie()` ?

Reply:
This condition is trying to handle the issue that a valid cookie is sent along with the request but no matched session in the database. In this case, I would prefer to redirect the user to log in again for session synchronization.